src/iuse.cpp

```diff
@@ -1901,63 +1902,84 @@ int iuse::mycus(player *p, item *it, bool t, const tripoint &pos)
+enum Petfood
+{
+    DOGFOOD,
+    CATFOOD,
+    CATTLEFODDER
+};
 
-int petfood(player *p, item *it, bool is_dogfood)
+int petfood( player *p, item *it, Petfood animal_food_type )
 {
     tripoint dirp;
-    if (!choose_adjacent(string_format(_("Put the %s where?"), it->tname().c_str()), dirp)) {
+    if( !choose_adjacent( string_format( _( "Put the %s where?" ), it->tname().c_str() ), dirp ) ) {
         return 0;
     }
     p->moves -= 15;
     const int mon_idx = g->mon_at( dirp, true );
-    if(mon_idx != -1) {
+
+    if( mon_idx != -1 ) {
         monster &mon = g->zombie( mon_idx );
-        if(mon.type->id == (is_dogfood ? mon_dog : mon_cat)) {
-            p->add_msg_if_player(m_good, is_dogfood
-                ? _("The dog seems to like you!")
-                : _("The cat seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with cats."));
-            mon.friendly = -1;
-            if( is_dogfood ) {
-                mon.add_effect( effect_pet, 1, num_bp, true );
-            }
-        } else if( is_dogfood && mon.type->id == mon_dog_thing ) {
-            p->deal_damage( &mon, bp_hand_r, damage_instance( DT_CUT, rng( 1, 10 ) ) );
-            p->add_msg_if_player( m_bad, _( "You want to feed it the dog food, but it bites your fingers!" ) );
-            if( one_in( 5 ) ) {
-                p->add_msg_if_player( _( "Apparently it's more interested in your flesh than the dog food in your hand!" ) );
-            }
-        } else {
-            p->add_msg_if_player( _( "The %s seems quite unimpressed!" ), mon.name().c_str() );
-        }
-    } else if( npc * const person_ = g->critter_at<npc>( dirp ) ) {
-        npc &person = *person_;
-        if( query_yn( is_dogfood ?
-            _( "Are you sure you want to feed a person the dog food?" ) :
-            _( "Are you sure you want to feed a person the cat food?" ) ) ) {
-            p->add_msg_if_player( _( "You put your %1$s into %2$s's mouth!" ), it->tname().c_str(), person.name.c_str() );
-            if( person.is_friend() || x_in_y( 9, 10 ) ) {
-                person.say( _( "Okay, but please, don't give me this again. I don't want to eat dog food in the cataclysm all day." ) );
-            } else {
-                p->add_msg_if_player( _( "%s knocks it out from your hand!" ), person.name.c_str() );
-                person.make_angry();
-            }
+
+        switch( animal_food_type ) {
+            case DOGFOOD:
+                if( mon.type->id == mon_dog_thing ) {
+                    p->deal_damage( &mon, bp_hand_r, damage_instance( DT_CUT, rng( 1, 10 ) ) );
+                    add_msg( m_bad, _( "You want to feed it the dog food, but it bites your fingers!" ) );
+                    if( one_in( 5 ) ) {
+                        add_msg(_( "Apparently it's more interested in your flesh than the dog food in your hand!" ) );
+                    }
+
+                } else if( npc *const person_ = g->critter_at<npc>( dirp ) ) {
```

Comment:
Unrelated to the PR: This condition can never be `true` (assuming no bug): the above `if` checks for a monster at `dirp`, which implies there can't be any other creature, so no npc will be there.